### PR TITLE
Choose upload version when package metadata are uploaded

### PR DESCRIPTION
**Pull request by olenagerasimova**
Part of #227 
Implemented choosing upload version on maven metadata upload: we first try to find suitable uploading snapshot, if there is not any, save metadata by release version. Also fixed several other problems:
- moving files to the repository after the update in `AstoMaven` and its test (because of the mistake in the test bug in `AstoMaven`) was not visible
- `AstoValidUpload` should check package metadata from the `meta` sub-key
- extended `DeployMetadata` with method to read snapshot versions
- enabled `deploysSnapshot` in `MavenITCase`

### CODE REVIEW

**Comment by olenagerasimova** [comment]
@g4s8 please, could you take a look at this PR?

File: src/main/java/com/artipie/maven/asto/AstoValidUpload.java [hunk 1]
```diff
@@ -83,7 +83,7 @@
      * Maven metadata and metadata checksums.
      */
     private static final Pattern PTN_META =
-        Pattern.compile(".+/maven-metadata.xml.(?:md5|sha1|sha256|sha512)");
+        Pattern.compile(".+/meta/maven-metadata.xml.(?:md5|sha1|sha256|sha512)");
```
Comment (g4s8) [inline src/main/java/com/artipie/maven/asto/AstoValidUpload.java:86]: @olenagerasimova why the pattern for metadata changed? I thought it's the standard path of Maven repository

File: src/main/java/com/artipie/maven/metadata/DeployMetadata.java [hunk 2]
```diff
@@ -57,4 +58,13 @@ public String release() {
             return release.get(0);
         }
     }
+
+    /**
+     * Reads snapshot versions from metadata.xml.
+     * @return List of snapshot versions
+     */
+    public List<String> snapshots() {
```
Comment (g4s8) [inline src/main/java/com/artipie/maven/metadata/DeployMetadata.java:66]: @olenagerasimova maybe it's better to return `Set` here to say that the snapshot versions doesn't contain duplicates?

**Review by g4s8 — CHANGES_REQUESTED** [review]
@olenagerasimova please check my comments and questions

File: src/main/java/com/artipie/maven/asto/AstoValidUpload.java (hunk 1 above)
Comment (olenagerasimova) [reply src/main/java/com/artipie/maven/asto/AstoValidUpload.java:86]: @g4s8 package metadata are now uploaded into `meta` subdirectory (check [readme](https://github.com/artipie/maven-adapter#upload-process)), this class works with our internal upload, that's why I've changed pattern.

**Comment by olenagerasimova** [comment]
@g4s8 thanks, corrected and answered

**Review by g4s8 — APPROVED** [review]
@olenagerasimova thanks

**Pull request merged**